client/src/main.rs

```diff
@@ -0,0 +1,239 @@
+use clap::Parser;
+use rustyline::Editor;
+use std::net::SocketAddr;
+use std::{io, net};
+use tokio::net::TcpListener;
+
+use fuel_vm::consts::{VM_MAX_RAM, WORD_SIZE};
+use fuel_vm::prelude::{Breakpoint, ContractId};
+
+use fuel_debugger::{Command, Response};
+
+use fuel_debugger_client::{names, Client, Listener};
+
+#[derive(Parser, Debug)]
+pub struct Opt {
+    #[clap(long = "ip", default_value = "127.0.0.1", parse(try_from_str))]
+    pub ip: net::IpAddr,
+
+    #[clap(long = "port", default_value = "4001")]
+    pub port: u16,
+}
+
+#[tokio::main]
+async fn main() -> io::Result<()> {
+    let config = Opt::parse();
+
+    let mut rl = Editor::<()>::new();
+
+    let listener = Listener::new(TcpListener::bind((config.ip, config.port)).await?);
+
+    println!(
+        "Listening for connections at {:?}",
+        SocketAddr::from((config.ip, config.port))
+    );
+
+    loop {
+        let (mut client, addr) = listener.accept().await?;
+        println!("Connected (remote {:?})", addr);
```

Comment:
use tracing instead of println for logging

Reply:
This is not logging, this is the user interface

Reply:
We have feedback messages that aren't the direct output of the CLI. Treating this as logging is beneficial also for pipe unix commands that expects only the output in stdout - and not health feedback.